rules/framework/vfs_overlay.bzl

```diff
@@ -1,40 +1,196 @@
-    if not files:
-        return
+# Make roots for a given framework. For now this is done in starlark for speed
+# and incrementality. For imported frameworks, there is additional search paths
+# enabled
+def _make_root(ctx, framework_name, root_dir, extra_search_paths, module_map, hdrs, private_hdrs, has_swift):
+    extra_roots = []
+    if extra_search_paths:
+        sub_dir = "Headers"
 
-    return {
-        "name": root_dir,
+        # Strip the build file path
+        base_path = "/".join(ctx.build_file_path.split("/")[:-1])
+        rooted_path = base_path + "/" + extra_search_paths + "/" + sub_dir + "/"
+
+        extra_roots = [{
+            "type": "file",
+            "name": file.path.replace(rooted_path, ""),
+            "external-contents": file.path,
+        } for file in hdrs]
+
+        extra_roots += [{
+            "type": "file",
+            "name": framework_name + "/" + file.path.replace(rooted_path, ""),
+            "external-contents": file.path,
+        } for file in hdrs]
+
+    modules_contents = []
+    if len(module_map):
+        modules_contents.append({
+            "type": "file",
+            "name": "module.modulemap",
+            "external-contents": module_map[0].path,
+        })
+
+    modules = []
+    if len(modules_contents):
+        modules = [{
+            "name": "Modules",
+            "type": "directory",
+            "contents": modules_contents,
+        }]
+
+    headers_contents = extra_roots
+    headers_contents.extend([
+        {
+            "type": "file",
+            "name": file.basename,
+            "external-contents": file.path,
+        }
+        for file in hdrs
+    ])
+
+    headers = []
+    if len(headers_contents):
+        headers = [{
+            "name": "Headers",
+            "type": "directory",
+            "contents": headers_contents,
+        }]
+
+    private_headers_contents = {
+        "name": "PrivateHeaders",
         "type": "directory",
         "contents": [
             {
                 "type": "file",
-                "name": override_name or file.basename,
+                "name": file.basename,
                 "external-contents": file.path,
             }
-            for file in files
+            for file in private_hdrs
         ],
     }
 
+    private_headers = []
+    if len(private_hdrs):
+        private_headers = [private_headers_contents]
+
+    roots = []
+    if len(headers) or len(private_headers) or len(modules):
+        roots += [{
+            "name": root_dir,
+            "type": "directory",
+            "contents": headers + private_headers + modules,
+        }]
+    if has_swift:
+        roots += [_vfs_swift_module_contents(ctx, framework_name, FRAMEWORK_SEARCH_PATH)]
+
+    return roots
+
+def _vfs_swift_module_contents(ctx, framework_name, root_dir):
+    # Forumlate the framework's swiftmodule - don't have the swiftmodule when
+    # creating with apple_library. Consider removing that codepath to make this
```

Comment:
maybe a link to the codepath will make future fixes easier? also trying to understand the context here

Reply:
Here it's referring to when this is leveraged by the `apple_library` macro - https://github.com/bazel-ios/rules_ios/blob/master/rules/library.bzl